File: domain/src/main/java/com/omvp/domain/interactor/impl/GetLocaleListUseCaseImpl.java
package com.omvp.domain.interactor.impl;

import com.omvp.domain.interactor.GetLocaleListUseCase;
import com.omvp.domain.repository.LocaleRepository;

import java.util.List;
import java.util.Locale;

import javax.inject.Inject;

import io.reactivex.Single;

public class GetLocaleListUseCaseImpl extends BaseUseCaseImpl<LocaleRepository> implements GetLocaleListUseCase {

    @Inject
    GetLocaleListUseCaseImpl(LocaleRepository repository) {
        super(repository);
    }

    @Override
    public Single<List<Locale>> execute() {
        return getRepository().retrieveList();
    }

}

File: domain/src/main/java/com/omvp/domain/interactor/impl/GetLocaleListUseCaseImpl.kt
package com.omvp.domain.interactor.impl

import com.omvp.domain.interactor.GetLocaleListUseCase
import com.omvp.domain.repository.LocaleRepository
import java.util.Locale

import javax.inject.Inject

import io.reactivex.Single

class GetLocaleListUseCaseImpl @Inject
internal constructor(repository: LocaleRepository) : BaseUseCaseImpl<LocaleRepository>(repository), GetLocaleListUseCase {

    override fun execute(): Single<List<Locale>> {
        return repository.retrieveList()
    }

}
